### PR TITLE
Added script and workflow to automatically update build artifact sizes in check_metadata.sh

### PR DESCRIPTION
**Pull request by gonfunko**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves
https://github.com/google/blockly/issues/4977

### Proposed Changes
Adds a script that calculates the size of the various Blockly JS files and updates check_metadata.sh with the new values.

#### Behavior Before Change
As part of the release process, a developer would need to manually determine the build artifact sizes and update the check_metadata.sh script.

#### Behavior After Change
Running the workflow will automatically calculate the new build artifact sizes, update the script, and create a PR with the updated script.

### Reason for Changes
Makes this step in the release process more reliable and reduces the effort on the part of the dev team.

### CODE REVIEW

**Review by cpcallen — COMMENTED** [review]
I _was_ going to suggest that you add a flag to `check_metadata.sh` to do this, rather than creating a separate script, but I've learned that [bash acts weirdly in the face of self-modifying shell scripts](https://stackoverflow.com/a/6303416/4969945) so you'd want to store the target sizes and history data in a separate file.  ([There are ways to make self-modifying bash scripts work safely](https://stackoverflow.com/a/2358432/4969945) but… yikes.)

File: tests/scripts/update_metadata.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+quarter=$(date "+Q%q %Y")
+version=$(npx -c 'echo "$npm_package_version"')
+
+replacement="# ${quarter}   ${version}    ${blockly_size}\nblockly_size_expected=${blockly_size}"
+sed -ri "s/blockly_size_expected=[0-9]+/${replacement}/g" tests/scripts/check_metadata.sh
+replacement="# ${quarter}   ${version}    ${blocks_size}\nblocks_size_expected=${blocks_size}"
+sed -ri "s/blocks_size_expected=[0-9]+/${replacement}/g" tests/scripts/check_metadata.sh
+replacement="# ${quarter}   ${version}    ${blockly_gz_size}\nblockly_gz_size_expected=${blockly_gz_size}"
+sed -ri "s/blockly_gz_size_expected=[0-9]+/${replacement}/g" tests/scripts/check_metadata.sh
+replacement="# ${quarter}   ${version}    ${blocks_gz_size}\nblocks_gz_size_expected=${blocks_gz_size}"
```
[the remark this answers is not in the document]
Comment (cpcallen) [reply tests/scripts/update_metadata.sh:19]: Oh, also: see [style guide re: line length and long string literals](replacement="# ${quarter} ${version} ${blockly_size}\nblockly_size_expected=${blockly_size}").

[the remark this answers is not in the document]
Comment (gonfunko) [reply tests/scripts/update_metadata.sh:19]: Yeah, I noticed the delimiter issue and decided to just go with the most recent, but having it be consistent SGTM, will hold off until that change is in.

**Comment by gonfunko** [comment]
I managed to clobber my branch, so I'm going to withdraw this PR and send a new one incorporating the changes you suggested.

**Comment by gonfunko** [comment]
https://github.com/google/blockly/pull/5033

**Pull request closed without merging**